src/Congo.RazorPages/Services/ProductsService.cs

```diff
@@ -28,80 +31,87 @@ public IEnumerable<Product> GetSampleProducts()
                 },
                 new()
                 {
-                    Id = 3,
+                    Id = Guid.NewGuid(),
                     Title = "A Rock",
                     Price = 0.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 4,
+                    Id = Guid.NewGuid(),
                     Title = "5-Piece Gardening Kit",
                     Price = 34.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 5,
+                    Id = Guid.NewGuid(),
                     Title = "Box of Chocolates",
                     Price = 9.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 6,
+                    Id = Guid.NewGuid(),
                     Title = "Window Cleaning Solution",
                     Price = 4.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 7,
+                    Id = Guid.NewGuid(),
                     Title = "Paper shredder",
                     Price = 44.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 8,
+                    Id = Guid.NewGuid(),
                     Title = "Xbox Controller",
                     Price = 119.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 9,
+                    Id = Guid.NewGuid(),
                     Title = "LED Light Bulb - 8 pack",
                     Price = 44.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
                 new()
                 {
-                    Id = 10,
+                    Id = Guid.NewGuid(),
                     Title = "A Dance With Dragons - Hard Cover",
                     Price = 39.99m,
                     ImageUrl = "https://via.placeholder.com/150"
                 },
             };
         }
 
-        public void PurchaseProduct(int id)
+        public async Task<Guid> Purchase(Guid productId)
         {
-            //
+            /* Uses the HttpClientFactory directly for now, will change it to use
+            Refit when it gets implemented. */
+            var client = _httpClientFactory.CreateClient(nameof(Congo));
+            var response = await client.PostAsync($"/api/{_productsUri}/{productId}/purchase", null);
+            var content = await response.Content.ReadAsStringAsync();
+            var result = JsonConvert.DeserializeObject<OrderConfirmationResponse>(content);
```

Comment:
```suggestion
            var result = await response.Content.ReadFromJsonAsync<OrderConfirmationResponse>();
```
There's a shorter way to read json responses